### ISSUE TITLE
Potential DOS attack by gossiped messages

### ISSUE THREAD

**Issue by twoeths** [ChainSafe/lodestar#5393]
**Describe the bug**

There are too many gossiped blocks sent to us with very old slot (out of the mcache windows and configured `seenTTL`)

```
grep -e "Received gossip block" -rn beacon-2023-04-21.log | grep "16Uiu2HAm3BTpUhsPgCPTYPJQ5EBH6DFTtVS7mMaKFioNYNChqrPi" | grep "curentSlot=6267911," | wc -l
122

grep -e "Received gossip block" -rn beacon-2023-04-21.log | grep "16Uiu2HAm3BTpUhsPgCPTYPJQ5EBH6DFTtVS7mMaKFioNYNChqrPi" | grep "curentSlot=6267919," | wc -l
133
```

We received more than 100 gossiped blocks per slot, ideally we should receive only 1!

some example logs
```
64910:Apr-21 01:02:41.674[network]       verbose: Received gossip block slot=6259233, root=0xb1fd…6653, curentSlot=6267911, peerId=16Uiu2HAm3BTpUhsPgCPTYPJQ5EBH6DFTtVS7mMaKFioNYNChqrPi, delaySec=104140.46600008011, recvToVal=2.2079999446868896
64911:Apr-21 01:02:41.690[network]       verbose: Received gossip block slot=6264267, root=0x950c…c3a6, curentSlot=6267911, peerId=16Uiu2HAm3BTpUhsPgCPTYPJQ5EBH6DFTtVS7mMaKFioNYNChqrPi, delaySec=43732.46900010109, recvToVal=2.2209999561309814
64912:Apr-21 01:02:41.708[network]       verbose: Received gossip block slot=6266950, root=0x1d9b…65c3, curentSlot=6267911, peerId=16Uiu2HAm3BTpUhsPgCPTYPJQ5EBH6DFTtVS7mMaKFioNYNChqrPi, delaySec=11536.471999883652, recvToVal=2.2360000610351562
```

right now we only ignore them after the deserialization which take our cpu time

**Expected behavior**

- Quick check the message slot, ignore them it's it's too old before sending to our queue
- Apply for all topics, not just `beacon_block`
- Right now the peer sending us these messages are identified as "Unknown", we should print agent id instead
- Inform client team if it happens frequently, it's likely a gossip issue, we only want to gossip messages in the first 3 windows of the mcache (3 * 0.7 = 2.1s)
- Potentially penalize that peers?

**Comment by twoeths** [comment]
Ideally we should receive 1 block per slot instead of this

<img width="1302" alt="Screenshot 2023-04-21 at 10 39 25" src="https://user-images.githubusercontent.com/10568965/233535393-c6b1aafb-76d7-4dcd-9e4f-f28b51b2929f.png">

**Comment by twoeths** [comment]
after the fix, our node process exactly 1 block per slot

<img width="1303" alt="Screenshot 2023-04-22 at 10 51 52" src="https://user-images.githubusercontent.com/10568965/233760792-0eb45c7d-cda8-450e-9410-493ceaa4ef72.png">

**Comment by philknows** [comment]
Closing unless there's anything outstanding after fix from #5394
